### PR TITLE
Feature/112820 recurrence rule form

### PR DESCRIPTION
**Pull request by nealfennimore**
https://central.tri.be/issues/112820

- Update withBlockClose HOC to export namespace and intercept functions
- Add in PreventBlockClose component to common for child components rendered outside of block to prevent closing of whole block
- Add in Select form component to common for select-like interactions
- Add in initial panel interactions for recurring

### CODE REVIEW

File: plugins/events-pro/src/modules/blocks/event-recurring/template.js [hunk 1]
```diff
@@ -13,40 +13,75 @@ import RecurringForm from '@moderntribe/events-pro/elements/recurring-form/eleme
 import ExceptionForm from '@moderntribe/events-pro/elements/exception-form/element';
 import RecurringAddField from '@moderntribe/events-pro/elements/recurring-add-field/element';
 import ExceptionAddField from '@moderntribe/events-pro/elements/exception-add-field/element';
+import Panel from '@moderntribe/events-pro/elements/panel/element';
 
 export default class EventRecurring extends PureComponent {
```
[the remark this answers is not in the document]
Comment (mitogh) [reply plugins/events-pro/src/modules/blocks/event-recurring/template.js]: 👍

File: plugins/events-pro/src/modules/elements/recurring-form/element.js [hunk 2]
```diff
@@ -3,16 +3,73 @@
...
+							/>
+						</div>
+					</div>
+					<div className="tribe-events-pro-recurring-block__row">
+						<div className="tribe-events-pro-recurring-block__row__label">
+							<span>{ __( 'On', 'events-gutenberg' ) }</span>
+							<span className="tribe-events-pro-recurring-block__row__label__bar">&nbsp;</span>
+						</div>
+						<div>
+							<DatePicker datetime="1999-01-01 12:00:00" />
```
[the remark this answers is not in the document]
Comment (mitogh) [reply plugins/events-pro/src/modules/elements/recurring-form/element.js]: 👍

**Comment by nealfennimore** [comment]
Good for more :eyes:

File: plugins/events-pro/src/modules/elements/recurring-form/element.js [hunk 3]
```diff
@@ -3,16 +3,73 @@
...
+						</div>
+						<div className="tribe-events-pro-recurring-block__row__time">
+							<TimePicker />
+							<span>{ __( 'to', 'events-gutenberg' ) }</span>
+							<TimePicker />
+							<div>
+								<ToggleControl
+									className="bob"
+									label={ __( 'Multi-Day', 'events-gutenberg' ) }
+									checked={ false }
```
[the remark this answers is not in the document]
Comment (nealfennimore) [reply plugins/events-pro/src/modules/elements/recurring-form/element.js]: WIP :)

File: plugins/events-pro/src/modules/blocks/event-recurring/template.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import React, { PureComponent } from 'react';
+import React, { PureComponent, Fragment } from 'react';
```
Comment (paulmskim) [inline plugins/events-pro/src/modules/blocks/event-recurring/template.js]: looks like you're not actually using the fragment here anymore?

**Pull request merged**